extensions-core/google-extensions/src/main/java/org/apache/druid/storage/google/GoogleDataSegmentKiller.java

```diff
@@ -52,7 +52,7 @@ public void kill(DataSegment segment) throws SegmentLoadingException
     Map<String, Object> loadSpec = segment.getLoadSpec();
     final String bucket = MapUtils.getString(loadSpec, "bucket");
     final String indexPath = MapUtils.getString(loadSpec, "path");
-    final String descriptorPath = indexPath.substring(0, indexPath.lastIndexOf('/')) + "/descriptor.json";
+    final String descriptorPath = indexPath.substring(0, indexPath.lastIndexOf('/')) + "/descriptor.json"; //-V6009
```

Comment:
What does `-V6009` mean?

Reply:
Refactored code to avoid the warning in the first place

Reply:
This inspection means "substring() argument can be negative".